operator_ui/src/pages/Jobs/New.test.tsx

```diff
@@ -108,7 +109,49 @@ describe('pages/Jobs/New', () => {
       target: { value: expected, name: 'jobSpec' },
     })
 
-    expect(getPersistJobSpec()).toContain(expected)
+    expect(storage.get(`${PERSIST_SPEC}${JobSpecFormats.JSON}`)).toContain(
+      expected,
+    )
+  })
+
+  it('persists spec definitions when switch the format', async () => {
+    const expectedJson = '{"foo":"bar"}'
+    const expectedToml = '"foo":"bar"'
+
+    const wrapper = mountWithProviders(
+      <Route path="/jobs/new" component={New} />,
+      {
+        initialEntries: [`/jobs/new`],
+      },
+    )
+
+    const textArea = wrapper.find('textarea[name="jobSpec"]')
+
+    textArea.simulate('change', {
+      target: { value: expectedJson, name: 'jobSpec' },
+    })
+
+    // switch to TOML
```

Comment:
Redundant. If you want to make the intent clearer then I'd prefer something along the lines of
```
const switchTo = (format) => {
  wrapper.find(`input[value="${format}"]`).simulate('change', {
      target: { value: format },
    })
}

...

switchTo(JobSpecFormats.TOML)
```

Reply:
Cheers! Spec updated.